frame/dapps-staking/src/migrations.rs

```diff
@@ -1077,3 +1077,169 @@ pub mod v3 {
         Ok(())
     }
 }
+
+pub mod festival_end {
+
+    use super::*;
+    use frame_support::{log, storage::child::KillStorageResult, traits::Get, weights::Weight};
+    use sp_runtime::traits::Zero;
+
+    #[cfg(feature = "try-runtime")]
+    pub fn pre_migrate<T: Config>() -> Result<(), &'static str> {
+        assert!(Ledger::<T>::iter().count().is_zero());
+        assert!(BlockRewardAccumulator::<T>::exists());
+
+        Ok(())
+    }
+
+    /// Should be executed during runtime upgrade
+    /// Won't do the entire cleanup, instead it will just do necessary preparatory actions.
+    pub fn on_runtime_upgrade<T: Config>(weight_limit: Weight) -> Weight {
+        for _ in Ledger::<T>::iter() {
```

Comment:
just curious - this method of checking is `Ledger` empty is used because there's no `is_empty` or similar trait on StorageMap, and is more efficient that `count()` (since it returns after first item is found)? Is there some other reason?

Reply:
Yes, I did it for the reasons you mentioned :)

Since iterator is lazy, this seems to be best way to check from what I've found.